### DISCUSSION TITLE
Easier way to create Filament user during development

### DISCUSSION THREAD

**Discussion by minthemiddle** [filamentphp/filament#296] [Ideas]
While developing and running `php artisan migrate:fresh` every 10 minutes to get an updated database structure, it's quite unpleasant to run `php artisan make:filament-user` and go through the full process again and again. And honestly, also answering the question "Do you want to star this repo?" gets annoying soon.

Two ideas:
- Give an option to run filament user creation as a factory, so I can add it to the database seeder with a preset user/password
- Give an argument for the command-line to at least disable the repo reminder and maybe also an option to pass user, email and pw directly

Or did  I miss a way to programmatically create user/password for Filament?

**Comment by ryangjchandler** [comment]
@minthemiddle The Filament user is just a normal Laravel model, there's no reason you can't use it in your factory or seeder like you would any other model. 

`Filament\Models\User` is the class you're looking for.

**Reply by ryangjchandler** [reply]
Also, I agree. If you say `No` to starring the repo, we shouldn't be asking you again. 

I'll open a PR for this tomorrow or the weekend.

**Reply by danharrin** [reply]
The user only gets asked if this is their first account... otherwise theres not a very obvious place to store the result of this check.

**Reply by ryangjchandler** [reply]
I think that's the problem though. If you wipe the database, you'll be asked again when you run `make:filament-user`. 

Simpler way of doing it is just writing a file to `storage/framework` and checking if it exists before asking the question again. Pretty sure Livewire does something similar.

**Reply by danharrin** [reply]
> Pretty sure Livewire does something similar. 

Nah AFAIK Livewire just checks if you already have a component class or not.

**Comment by lubikx** [comment]
Quick and dirty solution directly in seeder:

```php
<?php

namespace Database\Seeders;

use Filament\Commands\MakeUserCommand as FilamentMakeUserCommand;
use Illuminate\Database\Seeder;
use Illuminate\Support\Facades\Hash;

class DatabaseSeeder extends Seeder
{
    /**
     * Seed the application's database.
     */
    public function run(): void
    {
        $filamentMakeUserCommand = new FilamentMakeUserCommand();
        $reflector = new \ReflectionObject($filamentMakeUserCommand);

        $getUserModel = $reflector->getMethod('getUserModel');
        $getUserModel->setAccessible(true);
        $getUserModel->invoke($filamentMakeUserCommand)::create([
            'name' => 'YOUR NAME',
            'email' => 'YOUR@EMAIL.COM',
            'password' => Hash::make('YOUR-PASSWORD'),
        ]);
    }
}
```

This will get invoked when you clean your database, run migrations and seed the database, for example like this:
```
php artisan migrate:fresh --seed
```

**Reply by danharrin** [reply]
The make:filament-user command is literally just creating a user model, it doesn't do anything else special. You can just use the UserFactory. The purpose of the command is so people can get started quickly, even if they don't have a registration feature in their app.

**Reply by RattanakUNG** [reply]
Simply create user in the to the seeder is fine:
        User::factory()->create([
            'name' => 'yr_name',
            'email' => 'yr_email@domain.com',
            'email_verified_at' => now(),
            'password' => static::$password ??= Hash::make('password'),
        ]);
